extensions/eventtap/internal.m

```diff
@@ -173,12 +172,18 @@ static int eventtap_start(lua_State* L) {
                               eventtap_callback,
                               e);
 
-    CGEventTapEnable(e->tap, true);
-    e->runloopsrc = CFMachPortCreateRunLoopSource(NULL, e->tap, 0);
-    CFRunLoopAddSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
+    if (e->tap) {
+        e->running = true;
 
-    lua_settop(L,1);
-    return 1;
+        CGEventTapEnable(e->tap, true);
+        e->runloopsrc = CFMachPortCreateRunLoopSource(NULL, e->tap, 0);
+        CFRunLoopAddSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
+
+        lua_settop(L,1);
+        return 1;
+    } else {
+        showError(L, "Unable to create eventtap.  Is Accessibility enabled?");
+    }
```

Comment:
Yeah I think we should always try to return the object if possible, being able to chain the method calls is very pleasing. I would welcome an audit of the existing modules, and would definitely help out with such an effort :D